lib/tasks/demo.rake

```diff
@@ -869,4 +902,78 @@ namespace :demo do
   def create_fake_timestamp(from, to)
     Faker::Time.between_dates(from: from, to: to >= Date.today ? Time.now : to, period: :all)
   end
+
+  def duplicate_timestamps(from, to)
+    to.created_at = from.created_at
+    to.updated_at = from.updated_at
+  end
+
+  # Duplicate patient and all nested relations and change last name
+  def deep_duplicate_patient(patient, responder_id = nil)
+    new_patient = patient.dup
+    new_patient.responder_id = responder_id unless responder_id.nil?
+    new_patient.last_name = "#{Faker::Name.last_name}#{rand(10)}#{rand(10)}"
+    new_patient.submission_token = new_patient.new_submission_token
+    duplicate_timestamps(patient, new_patient)
+    new_patient.save
+    new_patient.update(responder_id: new_patient.id) if responder_id.nil?
+    patient.dependents.each do |p|
+      if p.id != p.responder_id
+         deep_duplicate_patient(p, new_patient.id)
+      end
+    end
+    patient.assessments.each do |assessment| 
+        new_assessment = assessment.dup
+        duplicate_timestamps(assessment, new_assessment)
+        rep_condition = assessment.reported_condition
+        new_reported_condition = rep_condition.dup
+        new_reported_condition.save
+        symptoms = []
+        assessment.reported_condition.symptoms.each do |s|
+            news = s.dup
+            duplicate_timestamps(s, news)
+            news.condition_id = new_reported_condition.id
+            symptoms << news
+        end
+        Symptom.import symptoms
+        new_assessment.patient_id = new_patient.id
+        new_assessment.save
+        new_reported_condition.update(assessment_id: new_assessment.id, updated_at: rep_condition.updated_at, created_at: rep_condition.created_at)
+    end
+
+    histories = []
+    patient.histories.each do |history| 
+        new_history = history.dup
+        duplicate_timestamps(history, new_history)
+        histories << new_history
+    end
+    History.import histories
+
+    patient.transfers.each do |transfer| 
+        new_transfer = transfer.dup
+        duplicate_timestamps(transfer, new_transfer)
+        new_transfer.patient_id = new_patient.id
```

Comment:
does this need to be done for the other associations as well? (histories, close_contacts, contact_attempts) maybe that can just be combined with the duplicate_timestamps method? I think it might also be faster to use `import` like you did for `History` because that will be 1 query instead of n